### PR TITLE
[dv/pwrmgr] Enable xcelium

### PR DESCRIPTION
**Pull request by matutem**
Make necessary changes to enable xcelium simulator.

Signed-off-by: Guillermo Maturana <maturana@google.com>

### CODE REVIEW

**Review by cindychip — APPROVED** [review]

File: hw/ip/pwrmgr/dv/env/seq_lib/pwrmgr_base_vseq.sv [hunk 1]
```diff
@@ -138,72 +138,42 @@ class pwrmgr_base_vseq extends cip_base_vseq #(
...
+    `uvm_info(`gfn, $sformatf("Driving %0s to %b", rsp_name, req), UVM_MEDIUM) \
+    drop_objection();
+
+  `define SLOW_RESPONSE_ON_EDGES(rsp_name, cycles, req, rsp) \
+      forever @(posedge req) begin \
+        `SLOW_RESPONSE_ON_EDGE(rsp_name, cycles, req, rsp) \
+      end \
+      forever @(negedge req) begin \
+        `SLOW_RESPONSE_ON_EDGE(rsp_name, cycles, req, rsp) \
+      end
```
Comment (weicaiyang) [inline hw/ip/pwrmgr/dv/env/seq_lib/pwrmgr_base_vseq.sv:161]: looks like we can merge 2 forever loops
```suggestion
      forever @(req) begin \
        `SLOW_RESPONSE_ON_EDGE(rsp_name, cycles, req, rsp) \
      end
```

File: hw/ip/pwrmgr/dv/env/pwrmgr_env_pkg.sv [hunk 2]
```diff
@@ -25,7 +25,7 @@ package pwrmgr_env_pkg;
   parameter string LIST_OF_ALERTS[] = {"fatal_fault"};
 
   // types
-  typedef struct {
+  typedef struct packed {
```
Comment (weicaiyang) [inline hw/ip/pwrmgr/dv/env/pwrmgr_env_pkg.sv:28]: is this the place Xcelium complains? the other update doesn't seem related to enable xcelium.

File: hw/ip/pwrmgr/dv/env/seq_lib/pwrmgr_base_vseq.sv [hunk 3]
```diff
@@ -212,11 +182,44 @@ class pwrmgr_base_vseq extends cip_base_vseq #(
...
+
+  `define FAST_RESPONSE_ON_EDGES(rsp_name, cycles, req, rsp) \
+      forever \
+        @(posedge req) begin \
+          `FAST_RESPONSE_ON_EDGE(rsp_name, cycles, req, rsp) \
+        end \
+      forever \
+        @(negedge req) begin \
+          `FAST_RESPONSE_ON_EDGE(rsp_name, cycles, req, rsp) \
+        end
```
Comment (weicaiyang) [inline hw/ip/pwrmgr/dv/env/seq_lib/pwrmgr_base_vseq.sv:204]: ditto

**Review by udinator — APPROVED** [review]

File: hw/ip/pwrmgr/dv/env/pwrmgr_env_pkg.sv (hunk 2 above)
Comment (matutem) [reply hw/ip/pwrmgr/dv/env/pwrmgr_env_pkg.sv:28]: You got it

File: hw/ip/pwrmgr/dv/env/seq_lib/pwrmgr_base_vseq.sv (hunk 1 above)
Comment (matutem) [reply hw/ip/pwrmgr/dv/env/seq_lib/pwrmgr_base_vseq.sv]: Thanks :)
Done.

File: hw/ip/pwrmgr/dv/env/seq_lib/pwrmgr_base_vseq.sv (hunk 3 above)
Comment (matutem) [reply hw/ip/pwrmgr/dv/env/seq_lib/pwrmgr_base_vseq.sv]: Done

File: hw/ip/pwrmgr/dv/env/seq_lib/pwrmgr_base_vseq.sv [hunk 4]
```diff
@@ -138,72 +138,42 @@ class pwrmgr_base_vseq extends cip_base_vseq #(
...
+                              cfg.pwrmgr_vif.slow_cb.pwr_ast_rsp.core_clk_val)
+      `SLOW_RESPONSE_ON_EDGES("io_clk_val", cycles_before_io_clk_en,
+                              cfg.pwrmgr_vif.slow_cb.pwr_ast_req.io_clk_en,
+                              cfg.pwrmgr_vif.slow_cb.pwr_ast_rsp.io_clk_val)
+      `SLOW_RESPONSE_ON_EDGES("usb_clk_val", cycles_before_usb_clk_en,
+                              cfg.pwrmgr_vif.slow_cb.pwr_ast_req.usb_clk_en,
+                              cfg.pwrmgr_vif.slow_cb.pwr_ast_rsp.usb_clk_val)
+      `SLOW_RESPONSE_ON_EDGES("main_pok", cycles_before_main_pok,
+                              cfg.pwrmgr_vif.slow_cb.pwr_ast_req.main_pd_n,
+                              cfg.pwrmgr_vif.slow_cb.pwr_ast_rsp.main_pok)
```
[the remark this answers is not in the document]
Comment (matutem) [reply hw/ip/pwrmgr/dv/env/seq_lib/pwrmgr_base_vseq.sv]: Thanks
